zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/ExecRemoteInterpreterProcess.java

```diff
@@ -175,6 +177,7 @@ public void waitForShutdown(int timeout) {
 
     @Override
     public void waitForReady(int timeout) {
+      this.waitForThread = Thread.currentThread();
       synchronized (this) {
```

Comment:
Wouldn't it be better to be inside the synchronized block?
And is it okay to not assign `this.waitForThread` to null again?

Reply:
Fixed